src/benchmark.rs

```diff
@@ -173,3 +199,59 @@ fn fetch_configured_refs(repo: &git2::Repository, refs: &[String]) -> Result<Pat
     }
     Ok(wd)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use core::panic;
+    #[tokio::test]
+    async fn test_resolve_env() {
+        let resolved_envs =
+            resolve_env_from_stdout(Command::new("echo").arg("[\"env0\", \"env1\", \"env2\"]"))
+                .await;
+        match resolved_envs {
+            Ok(v) => {
+                assert_eq!(v[0], "env0");
+                assert_eq!(v[1], "env1");
+                assert_eq!(v[1], "env1");
+            }
+            _ => panic!("Parsing unexpectedly failed for echo-ing a list of strings."),
+        }
+    }
+
+    #[tokio::test]
+    async fn test_resolve_env_empty_json() {
+        let resolved_envs = resolve_env_from_stdout(Command::new("echo").arg("[]")).await;
+        match resolved_envs {
+            Ok(v) => {
+                assert_eq!(v.len(), 0);
+            }
+            _ => panic!("Parsing unexpectedly failed for echo-ing an empty list."),
+        }
+    }
+
+    #[tokio::test]
+    async fn test_resolve_env_crash_integer_list() {
+        let resolved_envs = resolve_env_from_stdout(Command::new("echo").arg("[1, 2, 3]")).await;
+        match resolved_envs {
+            Ok(_) => panic!("Integer list is not an expected type"),
+            Err(e) => {
+                assert_eq!(
+                    format!("{e:?}"),
+                    "invalid type: integer `1`, expected a string"
+                );
+            }
+        }
```

Comment:
Rust has soo many combinators for `Result` and `Option`. One can spend far too much time to find the perfect one, but it’s often possible to avoid a verbose `match` with a cursory look at the docs.

```suggestion
        let e = resolved_envs.expect_err("Integer list is not an expected type");
        assert_eq!(
            format!("{e:?}"),
            "invalid type: integer `1`, expected a string"
        );
```